### DISCUSSION TITLE
Tagesertrag (yield day) wird mehrmals am Tag auf 0 gesetzt?

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#248] [General]
Ich stelle Mal hier die Frage: Kann es sein, dass der Wechselrichter mehrmals am Tag ein Reset macht?

ich verfolgte die Tages Produktion und merkte, der Tagesertrag hat sich mehrmals am Tag auf null gesetzt (siehe Bild).
![Neustart des Wechselrichters vielleicht](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/6b8a6538-844a-4125-a460-d473e5549e7f)

**Comment by helgeerbe** [comment]
Ja, der Wechselrichter wird über die DC Spannung betrieben. Wenn nicht genügend Leistung da ist, schaltet er ab. Wenn er wieder anläuft sind die Tageszähler genullt.

**Comment by madmartin** [comment]
Warscheinlich schaltet sich der WR aus, wenn die Eingangsspannung zu niedrig ist. Dann erholt sich die Batterie nach einer Weile, die Spannung steigt wieder etwas, und dann schaltet sich der WR wieder ein.

Daß dabei der Tageszähler immer wieder bei null anfängt ist eigentlich nur ein Schönheitsfehler.

Das wahre Problem in dieser Situation ist: das ist die beste Methode die Batterie kaputtzukriegen. Wiederholte Tiefentladung ist äußerst schädlich.

Wenn also die Batterie KEIN Managentsystem hat, daß die Abschaltung verursacht, dann mußt Du UNBEDINGT was an den Parametern des "Dynamic Power Limiter" ändern!

**Comment by Manos1966** [comment]
Das waere die logische Erklaerung.... 
Wie das Bild aber zeigt, die Spannung der Batterie (2 x 25,6V Batterien), blieb konstant.
Auch als die Batterie Spannung gegen 14:00 runter auf  44V ging, war es kein Problem.
![Neustart des Wechselrichters vielleicht](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/e394e045-034f-4bce-99df-d7ee113de3e8)

**Comment by helgeerbe** [comment]
Siehst du denn im LiveView in der openDTU im Event Log vom Invert den letzten Neustart?

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/59169507/577d5aa8-54c7-493a-996e-bb374cd897c1)

Versuche im Anschluss mal über die openDTU Oberfläche den Inverter neu zu starten. Ist das Verhalten dann für ein paar Tage wieder OK? Ich habe so das Gefühl, dass der Inverter Probleme machen kann, wenn er ohne Neustart dauernd läuft. Kann das aber nicht mit Messdaten belegen.

**Comment by madmartin** [comment]
Ich bin ja gerade dabei einen konfigurierbaren täglichen Neustart des Inverter/Wechselrichter zu implementieren. ist bald fertig.

**Comment by Manos1966** [comment]
@helgeerbe I did not find anything in the Event Log. 
I need to follow it up for some days...
**and** I need to recalculate the "DC Spannung - Stop" with the Lastkorrekturfaktor (maybe you are right and it is the Battery Limit that stops the inverter).
On the other side, it worked perfectly today!
![Neustart des Wechselrichters vielleicht2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/94e799c0-3e9c-44b4-99bc-b2e07cc52e97)
(the pink line is the second inverter we have working in parallel with two panels)

**Reply by spcqike** [reply]
kannst du dein Setup mal näher beschreiben? wie ist was angeschlossen? was für Leitungen/Querschnitte/Längen, wo misst du welche Spannungen/Werte?
1.500W bei 50V sind ja 30A, Wenn da was nicht korrekt passt, hast du unter Umständen am Eingang des Inverters einen riesigen Spannungsabfall, was den Unterspannungsschutz auslösen könnte. Aber ohne genauer Beschreibung deines Aufbaus ist das alles nur raten :)

**Reply by Manos1966** [reply]
@spcqike danke fuer die Vorschläge, ich werde es mal pruefen.
Vorab aber, es ist zwar (erfreulich) überraschend, es scheint dem HOYMILES HM-1500 (vier Eingaenge, zwei MPPT) nicht zu stoeren, wie das Bild zeigt:
![HM-1500 at max](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/13d64a5f-e591-4803-bf86-2c9ebfa478b3)
Alle vier HOYMILES Eingaenge (Strings) sind an die Batterie angeschlossen.

**Reply by spcqike** [reply]
der HM erkennt eine Spannung von 45V. Ist das die Spannung, die du oben in FHEM protokolliert hast?
Kannst du die Spannung direkt an der Batterie nachmessen? 

Zusätzlich zum Einbruch der Spannung in der Batterie bei Belastung, hast du ja noch den Spannungsabfall über den Zuleitungen. Daher auch die Frage, was für Kabel du da hast und wie das genau aufgebaut ist. Im Worstcase hast du einfach 1x 6mm² von der Batterie zum WR und hast dort mit Y-Kabeln aufgeteilt auf die Eingänge. Besser wären natürlich einzelne Zuleitungen bzw. ein dickeres Kabel. Zum Innenwiderstand des Leiters kommen ja noch alle Steck- und Klemmverbindungen. Da wären, je nach Aufbau, 3-5V Spannungsabfall rein durch die Zuleitung nicht ungewöhnlich. (wenn auch nicht schön, da dies bei fast 36A immerhin 108-180W Verlustleistung bedeutet)

**Comment by kloppy1984** [comment]
Bei mir steht sehr oft im log: Shut down by remote control.

Gefühl passiert das wenn viel hoch und runter geregelt wird. Vor allem kurz vor dem ins Bett gehen,  Licht an,  Licht aus usw... 
Spannung hat nie die eingestellte min Grenze erreicht. 
Alle Daten sehr ich schön in Grafana...

**Reply by Manos1966** [reply]
We are talking about this eventlog, right? I have no messages...
I receive a lot of "ugly" data, values far above normal limit. These are "cleaned up" (removed) from the graphs.
I suspect the reason is because there is a total of **three** DTUs connected to the HOYMILES (one OpenDTUonBattery, one OpenDTU, one Hoymiles DTU) currently so, I considered the strange values "normal" - We are testing the OpenDTUonBattery, as soon as we have a stable configuration, we will remove the other DTUs and see if the data quality is better.  
![Eventlog](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/73d7accc-8db4-461a-8aeb-e823525caf76)

**Reply by spcqike** [reply]
wo kommt dieses Bild her? Hier hast du 5 Meldungen auf dem Inverter und 12Wh Tagesertrag, also bei der gegebenen Leistung von 1475W knappe 29sek Einspeisung.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/22fad23a-5c47-4dd1-ac6c-878248f47d14)

**Reply by Manos1966** [reply]
Das Bild mit den 5 Meldungen ist 2 Wochen alt (als wir die Max Last des HM-1500 getestet haben).
Das Bild mit null Ereignisse ist von heute morgen.

**Reply by kloppy1984** [reply]
opdendtu Battery will never run stable if other DTUs are watching to the same hoymiles!

**Comment by Manos1966** [comment]
Eine Frage habe ich noch:

> Korrigierte Spannung = DC Spannung + (Aktuelle Leistung (W) + Korrekturfaktor). 

By the way, I presume you wanted to write Aktuelle Leistung (W) **X** Korrekturfaktor)
Wenn der DC Spannung - Stop Wert erreicht wird, was passiert mit der Hoymiles? Wird sie **ausgeschaltet** und vielleicht dadurch der Tagesertrag auf null gesetzt?
![DC Spannung Stop](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ab8524da-4166-4e85-afe4-c2a60320806d)

**Reply by spcqike** [reply]
Ja es sollte wohl "mal Korrekturfaktor" sein. In deinem Fall, wenn dein Faktor von 0,0013 stimmt, hast du bei 1475W (von oben) ja 1,91V Spannungsfall. und damit immer noch 68W Verluste auf den Zuleitungen.

Zu deiner Frage:
Nein, ein Ausschalten des Wechselrichters per Software setzt den Tageszähler nicht zurück. Zumindest hab ich meinen im Interface grad aus gemacht und der Zähler ist so geblieben.

Der Tageszähler ist intern im Wechselrichter und setzt sich zurück, wenn der Wechselrichter seine Betriebsspannung (DC) verliert. Sei es, weil die Sonne auf den Paneelen fehlt oder weil die Batterie abgeklemmt wurde (was bspw. bei Solarladereglern passiert, wenn der Inverter dort am Lastausgang betrieben wird. Dazu gesellt sich dann das Problem der hohen Anlaufströme. Die Eingangskondensatoren ziehen dabei kurzzeitig alles, was geht, und könnten beschädigt werden.)

was spricht dagegen, den Gesamtertrag zu verwenden? Auch in FHEM sollte es möglich sein, basierend auf einem fortlaufend hochzählenden Wert die Differenz zwischen zwei Zeiträumen zu berechnen. bzw, inkremente und diese summiert pro Zeitraum X darzustellen.
https://forum.fhem.de/index.php/topic,84969.msg1207068.html#msg1207068

**Reply by Manos1966** [reply]
> Dazu gesellt sich dann das Problem der hohen Anlaufströme

du hast Recht, erfreulicherweise verhaelt sich der Hoymiles viel besser als wir erwartet haben. 
Es ist mir zwar unerklaerlich warum, aber die empirische Werte sind eben so, der HOYMILES regelt unter 10 A:
https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/244
Es ist eine Test-Installation, ich wollte eine Installation testen (bevor ich sie kaufe), seitdem bin ich nur am testen....

PS. Danke fuer den FHEM Link :-)

**Reply by spcqike** [reply]
du darfst die Stromregelung im Betrieb nicht mit dem Anlaufstrom beim Verbinden verwechseln.

Du kennst sicherlich die Funken, wenn man ein billiges Schaltnetzteil in die Steckdose steckt? das ist quasi das Gleiche.

Die leeren Kondensatoren am Eingang laden sich auf und können dabei Strom ziehen ohne Ende. dieser kurze Moment des Aufladens kann zur Beschädigung am Wechselrichter führen.
Wenn die Dinger erstmal aufgeladen sind, ist das relativ unproblematisch. im Regelbetrieb begrenzt der Wechselrichter sich von selbst. Sowohl in Abhängigkeit vom Strom als auch von der Temperatur.

Und: Im Normalbetrieb mit Paneelen kommt es gar nicht zu solchen hohen Strömen. Selbst wenn die Paneele mehr als 11,5A können, werden die Kondensatoren mit den ersten Sonnenstrahlen und mit den wenigen Watt/Ampere aufgeladen, die da fließen können.

**Reply by Manos1966** [reply]
Ich werde pruefen, ob die Batterien "Soft Start" haben.

**Comment by kloppy1984** [comment]
Ich habe zudem festgestellt, das der Wechselrichter im Standby geht, sobald der Victron in die Absorbtion geht

**Comment by Manos1966** [comment]
Just a short update, 
I changed the Battery DC Spannung - Stop to 48,5V (thanks to all members for their comments and above all, explanations on how these batteries behave).
This seems to have fixed the problem of the Tagesertrag going to zero many times during the day. The battery seems to behave nicely through the night, without resetting the daily production  values. 
Right now the battery is under heavy load (the test machine is at full power)  and the Voltage dropped to 47,5 V (the Panels Volt is actually Battery Volt as all four inputs are connected to the battery. It is still OK because under the current stress-load of 1471W : Korrigierte Spannung = DC Spannung + (Aktuelle Leistung (W) x Korrekturfaktor) it will stop at 46,6V)
The battery already delivered 2617Wh (2,6kWh) Tagesertrag so, it delivered its capacity.
![Delivery of up to 2,6kWh](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/1d3a89ca-9149-424d-a5f1-bfaff9b7b299)

**Reply by Manos1966** [reply]
Question @helgeerbe : I can not receive per MQTT the values of DC Spannung - Start, - Stop, - Lastkorrekturfaktor, am I right about that?
I wanted to create a real-time calculation of the Battery-Voltage Stop in FHEM.

**Reply by helgeerbe** [reply]
Your are right. Those values are currently not published. You could pull them via `http://opendtu/api/powerlimiter/status` as a workaround.

**Reply by Manos1966** [reply]
Awesome! 👍 
![VoltageThreshold](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ccfccd7d-2ca5-4edb-b429-16c479b0f141)

**Reply by Manos1966** [reply]
So now I have added the information and it seems to be working nicely (what I mean is that, using "simple" batteries it is manageable).
![Monitoring Battery](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/5bd8fdf4-2213-42b0-8bb2-374220a98495)
I hope the is a lot of sunshine today in order to see how the voltage data will be when the VICTRON starts charging the batteries and whether I should change the calculation of the battery capacity by taking into account whether VICTRON is charging or not.

Question @helgeerbe : If I have a battery with CANbus, will I be able to see the real capacity of the battery, even **while the VICTRON is charging it** ? 
I am referring to this sentence:

> Pylontech Battery support (via CAN bus interface). 
**Use the SOC for starting/stopping the power output** and provide the battery data via MQTT (autodiscovery for home assistant is currently not supported). Pin Mapping is supported (default RX PIN 27, TX PIN 26). Actual no live view support for Pylontech Battery.

**Reply by Manos1966** [reply]
as expected, VICTRON is on, gets 875W and 85,9V from three panels, and started charging the battery with 54V & 15,6A
The HOYMILES' inputs have 53,5V thus assume the battery is 100% full. 
This explains the erratic behaviour during the day when the inverter was going up and down and the sudden voltage drop (when each Battery BMS was shutting down the battery). [I am not complaining, I am just sharing information]

I need to experiment with the Solar Passthrough and the other settings (it is a bit difficult to understand it).
@helgeerbe **MAYBE** it would be useful to have an option like "Use at least xx% (e.g. 50%) of the VICTRON energy to charge the battery and the remaining xx% percent to have the HOYMILES produce energy"
With such an option, it is sure that the battery is charged during the day and it will not drain out.
_Probably you already have a similar possibility with the combination of SolarPassthrough and BAttery Strategy, I just need to understand it_

![Akku Kapazietaet while VICTRON](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/1e96e94f-ea4c-4076-90e0-28607e94e54f)
